projects/backend/src/main/java/backend/controllers/UsersController.java

```diff
@@ -1,37 +1,77 @@
 package backend.controllers;
 
-import org.springframework.web.bind.annotation.DeleteMapping;
+import backend.domain.User;
+import backend.dto.UserDto;
+import backend.service.UserService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+
+
+
 @RestController
 @RequestMapping("users")
 public class UsersController {
 
-  @PostMapping("signup")
-  public String signup() {
-    return "User Signup Endpoint";
-  }
+  private final UserService userService;
+  private final ModelMapper modelMapper;
+  private final BCryptPasswordEncoder passwordEncoder;
+
 
-  @PostMapping("login")
-  public String login() {
-    return "User Login Endpoint";
+  /**
+   * Constructor.
+   * @param userService user service
+   * @param modelMapper model mapper
+   * @param passwordEncoder password encoder
+   */
+  @Autowired
+  public UsersController(UserService userService,
+                         ModelMapper modelMapper,
+                         BCryptPasswordEncoder passwordEncoder) {
+    this.userService = userService;
+    this.modelMapper = modelMapper;
+    this.passwordEncoder = passwordEncoder;
   }
 
-  @DeleteMapping("logout")
-  public String logout() {
-    return "User Logout Endpoint";
+  /**
+   * User sign up.
+   * @param userDto user Data
+   * @return ResponseEntity
+   */
+  @PostMapping("signup")
+  public ResponseEntity signup(@RequestBody UserDto userDto) {
+    var user = modelMapper.map(userDto, User.class);
+    user.setPassword(passwordEncoder.encode(user.getPassword()));
+    userService.add(user);
+    return ResponseEntity.ok().build();
   }
 
-  @PutMapping("edit")
-  public String edit() {
-    return "User Edit Endpoint";
+  /**
+   * Editing user.
+   * @param userId id of user
+   * @param userDto data of user
+   * @return
+   */
+  @PutMapping("edit/{userId}")
+  public ResponseEntity edit(@PathVariable String userId,
+                             @RequestBody UserDto userDto) {
+    var user = modelMapper.map(userDto, User.class);
+    user.setId(userId);
+    userService.update(user);
+    return ResponseEntity.ok().build();
   }
 
   @PostMapping("forgot")
-  public String forgot() {
-    return "User Forgot Endpoint";
+  public ResponseEntity forgot() {
+    //TODO forgot pass logic
+    return ResponseEntity.ok().build();
```

Comment:
Cause it is not implemented it's invalid -> do not return ok, return something that signals error. 
Did you added a task ``` //TODO forgot pass logic``` to Trello?